libgnucash/engine/Account.cpp

```diff
@@ -3396,34 +3390,12 @@ xaccAccountGetCommodity (const Account *acc)
 
 gnc_commodity * gnc_account_get_currency_or_parent(const Account* account)
 {
-    gnc_commodity * commodity;
-    g_return_val_if_fail (account, nullptr);
+    g_return_val_if_fail (GNC_IS_ACCOUNT (account), nullptr);
+
+    for (auto acc = account; acc; acc = gnc_account_get_parent (acc))
+        if (auto comm = xaccAccountGetCommodity (acc); gnc_commodity_is_currency (comm))
+            return comm;
```

Comment:
std::find_if

Reply:
find_if wants c++ iterators though. this is acc climbing the parent tree.

Reply:
Oh, right.